src/main/java/com/guciowons/shoppingbasket/PriceRecord/PriceRecordController.java

```diff
@@ -0,0 +1,52 @@
+package com.guciowons.shoppingbasket.PriceRecord;
+
+import org.springframework.format.annotation.DateTimeFormat;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.time.LocalDateTime;
+import java.util.List;
+
+@RestController
+@RequestMapping("/pricerecords")
+public class PriceRecordController {
+    private final PriceRecordService priceRecordService;
+
+    public PriceRecordController(PriceRecordService priceRecordService) {
+        this.priceRecordService = priceRecordService;
+    }
+
+    @GetMapping
```

Comment:
To całe jest ogólnie dobrze, ale przez np łatwość użytkowania dla osób, które będą korzystać z tego API fajnie byłoby np zostąpić te 4 endpointy jednym -- zrób po prostu `/prices` które mają opcjonalne parametry na `from`, `to`, `productId` i w zależności do user poda to tak mu się to pofiltruje.

Lepiej jest się integrować z jednym endpointem niż z czterema ; )